ddtrace/opentelemetry/span.go

```diff
@@ -35,13 +37,34 @@ func (s *span) TracerProvider() oteltrace.TracerProvider        { return s.otelt
 func (s *span) AddEvent(_ string, _ ...oteltrace.EventOption)   { /*	no-op */ }
 func (s *span) RecordError(_ error, _ ...oteltrace.EventOption) { /*	no-op */ }
 
-func (s *span) SetName(name string) { s.SetOperationName(name) }
+func (s *span) SetName(name string) {
+	s.attributes[ext.SpanName] = strings.ToLower(name)
+}
 
 func (s *span) End(options ...oteltrace.SpanEndOption) {
 	if s.finished {
 		return
 	}
 	s.finished = true
+	for k, v := range s.attributes {
+		//	if we find operation.name,
+		if k == operationNameKey || k == ext.SpanName {
+			//	set it and keep track that it was set to ignore everything else
+			if name, ok := v.(string); ok {
+				s.attributes[ext.SpanName] = strings.ToLower(name)
+			}
+		}
+	}
```

Comment:
Refactoring suggestion:
Similar to the Java pseudocode in the doc, you can just check for an `operation.name` attribute in the converter function, and return out early if it exists.
If you do that, then this entire loop can disappear.

i.e. the only changes you need in this entire function then are:
```
	s.SetName(s.createOperationName())
	for k, v := range s.attributes {
		s.SetTag(k, v)
	}
```

Reply:
Sorry, what do you mean by `converter function` ?

Reply:
By "converter function" I meant the `createOperationName` method you have.